test/e2e/prune_test.go

```diff
@@ -580,4 +587,64 @@ var _ = Describe("Podman prune", func() {
 		Expect(err).ToNot(HaveOccurred())
 		Expect(dirents).To(HaveLen(3))
 	})
+
+	It("podman system prune --build clean up after terminated build", func() {
+		useCustomNetworkDir(podmanTest, tempdir)
+
+		podmanTest.BuildImage(pruneImage, "alpine_notleaker:latest", "false")
+
+		create := podmanTest.Podman([]string{"create", "--name", "test", BB, "sleep", "10000"})
+		create.WaitWithDefaultTimeout()
+		Expect(create).Should(ExitCleanly())
+
+		containerFilePath := filepath.Join(podmanTest.TempDir, "ContainerFile-podman-leaker")
+		err := os.WriteFile(containerFilePath, []byte(longBuildImage), 0755)
+		Expect(err).ToNot(HaveOccurred())
+
+		build := podmanTest.Podman([]string{"build", "-f", containerFilePath, "-t", "podmanleaker"})
+		// Build will never finish so let's wait for build to ask for SIGKILL to simulate a failed build that leaves stage containers.
+		// In case build never asks for SIGKILL, this will timeout and fail the test.
+		done := make(chan struct{})
+		go func() {
+			for {
+				if build.LineInOutputContains("Please use signal 9") {
+					break
+				}
+				time.Sleep(100 * time.Millisecond)
+			}
+			build.Signal(syscall.SIGKILL)
+			close(done)
+		}()
+		Eventually(done, defaultWaitTimeout).Should(BeClosed())
```

Comment:
This will still hang the goroutine which means you leak the entire goroutine. This really needs a timeout in the loop so the loop actually ends.

```go
matchedOutput := false

for range 900 {
	if build.LineInOutputContains("Please use signal 9") {
		matchedOutput = true
		break
	}
	time.Sleep(100 * time.Millisecond)
}
if !matchedOutput {
	Fail("Did not match special string in podman build")
}
```

Also when using goroutines in a ginkgo test one should use `defer GinkgoRecover()` to ensure error reporting there works. In this case it may not strictly be needed as there is no actual matching done but in case of a panic it should still be recovered by ginkgo.

Reply:
Thanks. I didn't realize that. Fixed.